src/components/AddItem.js

```diff
@@ -39,56 +40,32 @@ const AddItem = () => {
     getItems();
   }, []);
 
-  const cleanString = (str) => {
-    const regex = /[a-z]/g;
-    return str.trim().toLowerCase().match(regex).join('');
-  };
-
-  const validateInput = (list) => {
-    if (!list.itemName) {
-      setErrors(errorsList['empty']);
-      return false;
-    } else {
-      for (const item of items) {
-        if (cleanString(item.itemName) === cleanString(list.itemName)) {
-          setErrors(errorsList['duplicate']);
-          return false;
-        }
-      }
-    }
-    return true;
-  };
-
   const addItems = async () => {
     const refData = await getDocs(itemsCollectionRef);
     let tokenList = refData.docs.map(({ id }) => id);
 
-    const newList = {
-      itemName,
-      nextPurchase,
-      lastPurchase,
-      estimatedPurchaseInterval: null,
-      totalPurchases: 0,
-    };
-
     if (tokenList.includes(currToken)) {
-      if (validateInput(newList)) {
+      if (validateInput({ newItem, setErrors, errorsList, items })) {
         await updateDoc(doc(db, 'shopping-list', currToken), {
-          items: arrayUnion(newList),
+          items: arrayUnion(newItem),
         });
       } else {
         setHasError(true);
         setTimeout(() => setHasError(false), 3000);
       }
     } else {
-      if (validateInput(newList)) {
+      if (validateInput({ newItem, setErrors, errorsList, items })) {
         await setDoc(doc(db, 'shopping-list', currToken), {
-          items: [newList],
+          items: [newItem],
         });
       }
     }
   };
 
+  useEffect(() => {
+    addItems();
+  }, [newItem]);
+
```

Comment:
Eerm I am not sure why addItems is called in a `useEffect`. It runs immediately after a user gets to the Additem page and throws errors even before I get to add any items.

<img width="734" alt="Screenshot 2021-11-21 at 7 47 53 AM" src="https://user-images.githubusercontent.com/57858821/142752647-7ce7851a-68fd-4706-98c1-d82b7e1a6b3b.png">

You intended for it to run after the button is clicked right?